sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/DeadletterQueueSample.java

```diff
@@ -0,0 +1,255 @@
+
+        senderClient.close();
+    }
+
+    /**
+     * Send {@link ServiceBusMessage messages} to an Azure Service Bus Queue.
+     *
+     * @Param senderAsyncClient Service Bus Sender Client
+     * @Param maxMessages Maximum Number Of Messages
+     */
+    void sendMessages(ServiceBusSenderClient senderClient, int maxMessages) throws JsonProcessingException {
+        List<Person> messageList = Arrays.asList(
+            new Person("Einstein", "Albert"),
+            new Person("Heisenberg", "Werner"),
+            new Person("Curie", "Marie"),
+            new Person("Hawking", "Steven"),
+            new Person("Newton", "Isaac"),
+            new Person("Bohr", "Niels"),
+            new Person("Faraday", "Michael"),
+            new Person("Galilei", "Galileo"),
+            new Person("Kepler", "Johannes"),
+            new Person("Kopernikus", "Nikolaus")
+        );
+        for (int i = 0; i < Math.min(messageList.size(), maxMessages); i++) {
+            final String messageId = Integer.toString(i);
+            ServiceBusMessage message = new ServiceBusMessage(objectMapper.writeValueAsString(messageList.get(i)));
+            message.setContentType("application/json");
+            message.setSubject(i % 2 == 0 ? "Scientist" : "Physicist");
+            message.setMessageId(messageId);
+            message.setTimeToLive(Duration.ofMinutes(2));
+            System.out.printf("\tMessage sending: Id = %s%n", message.getMessageId());
+            senderClient.sendMessage(message);
+            System.out.printf("\tMessage acknowledged: Id = %s%n", message.getMessageId());
+        }
+    }
+
+    /**
+     * Receive {@link ServiceBusMessage messages} and return {@link ServiceBusMessage messages} back to the queue.
+     * When the time to life of the {@link ServiceBusMessage messages} expires,
+     * the {@link ServiceBusMessage messages} will be dumped as dead letters into the dead letter queue.
+     * We can receive these {@link ServiceBusMessage messages} from the dead letter queue.
+     *
+     * @Param connectionString Service Bus Connection String
+     * @Param queueName Queue Name
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    void deadLetterByExceedingMaxDelivery(String connectionString, String queueName) throws InterruptedException {
+        ServiceBusReceiverAsyncClient receiverAsyncClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .receiver()
+            .queueName(queueName)
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .buildAsyncClient();
+        receiverAsyncClient.receiveMessages().subscribe(receiveMessage -> {
+            System.out.printf("Picked up message; DeliveryCount %d%n", receiveMessage.getDeliveryCount());
+            // return message back to the queue
+            receiverAsyncClient.abandon(receiveMessage).subscribe();
```

Comment:
flatMap is probably better than concatMap? We're not aggregating multiple publishers.

Reply:
The main idea of this comment is to call the abandon method in the stream instead of in the subscribe method. I'm good with flatMap, which has hidden concurrency while concatMap doesn't.